sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/LookupCatalog.scala

```diff
@@ -106,24 +107,29 @@ private[sql] trait LookupCatalog extends Logging {
 
     def unapply(nameParts: Seq[String]): Option[(CatalogPlugin, Identifier)] = {
       assert(nameParts.nonEmpty)
-      if (nameParts.length == 1) {
-        Some((currentCatalog, Identifier.of(catalogManager.currentNamespace, nameParts.head)))
+      val (catalog, ident) = if (nameParts.length == 1) {
+        (currentCatalog, Identifier.of(catalogManager.currentNamespace, nameParts.head))
       } else if (nameParts.head.equalsIgnoreCase(globalTempDB)) {
         // Conceptually global temp views are in a special reserved catalog. However, the v2 catalog
         // API does not support view yet, and we have to use v1 commands to deal with global temp
         // views. To simplify the implementation, we put global temp views in a special namespace
         // in the session catalog. The special namespace has higher priority during name resolution.
         // For example, if the name of a custom catalog is the same with `GLOBAL_TEMP_DATABASE`,
         // this custom catalog can't be accessed.
-        Some((catalogManager.v2SessionCatalog, nameParts.asIdentifier))
+        (catalogManager.v2SessionCatalog, nameParts.asIdentifier)
       } else {
         try {
-          Some((catalogManager.catalog(nameParts.head), nameParts.tail.asIdentifier))
+          (catalogManager.catalog(nameParts.head), nameParts.tail.asIdentifier)
         } catch {
           case _: CatalogNotFoundException =>
-            Some((currentCatalog, nameParts.asIdentifier))
+            (currentCatalog, nameParts.asIdentifier)
         }
       }
+      if (CatalogV2Util.isSessionCatalog(catalog) && ident.namespace.isEmpty) {
```

Comment:
@cloud-fan For a session catalog, I could make this assumption that the namespace is required, right? I looked at `CatalogManager` that uses `v1SessionCatalog` for setting current namespace if the current catalog is a session catalog; and `v1SessionCatalog` requires the namespace (database) to already exist.

Reply:
shall we do this check inside `SessionCatalogAndIdentifier`?

Reply:
or inside `ResolveSessionCatalog`

Reply:
`CatalogAndIdentifier` should just focus on extracting catalog and identifier.

Reply:
The reason I put here is that `CatalogAndIdentifier` is also used here: https://github.com/apache/spark/blob/7c4ad6316e3fe4a58629771a99faaa986c866c0b/sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriterV2.scala#L53-L57, which means I have to put the same check everywhere `CatalogAndIdentifier` is used?

Reply:
I think `V2SessionCatalog` should stop filling the default database. It should assume the input identifier is the final identifier and fail it doesn't have database part. Then we don't need to do the check here.

Reply:
Good point, let me explore this route. Thanks!